### PR TITLE
nixos/nixos-enter: bind mount /etc/resolv.conf to chroot

### LINKED ISSUE #39665: (nixos-install & nixos-enter) Chroot cannot resolve dns names
## Issue description

After installing NixOS then using nixos-enter to chroot into the installation dns does not work.  I am trying to clone down a repository but it fails because it cannot resolve github.com.

Here is a picture of the of some commands run inside the chrooted installation
capture

[Bootstrap.sh](https://github.com/Wolfereign/nixos-configs/blob/8321bb6bbee9a1e38eabeaf81ccc920270f806c9/bootstrap.sh) being run to install and call the chrooted commands.
[Configuration.nix](https://github.com/Wolfereign/nixos-configs/blob/8321bb6bbee9a1e38eabeaf81ccc920270f806c9/machines/testing/configuration.nix) being used

### Steps to reproduce

1. Boot a vm using one of the 18.03.132155.b50443b5c4a isos (UEFI Mode).
2. Clone this [repo and tree](https://github.com/Wolfereign/nixos-configs/tree/8321bb6bbee9a1e38eabeaf81ccc920270f806c9)
3. run bootstrap.sh
4. select "test" machine (option 4) when prompted  (this is setup for hyper-v so you may need to fix the hardware-configuration.nix back to the default one generated).
5. Observer error after installation finishes

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.
capture2

### PR DESCRIPTION
**Pull request by Artturin**
###### Motivation for this change
Closes #39665

###### Things done

To test
```
gh pr checkout 137677
cd nixos
nix build -f default.nix -I nixos-config=modules/installer/cd-dvd/installation-cd-minimal.nix config.system.build.isoImage
```

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
@cole-h hi, this and some other PRs have ofborg-internal-error labels

**Comment by cole-h** [comment]
@ofborg eval

Been busy between work and school; logs look harmless, so just gonna remove the labels. Thanks!

File: nixos/modules/installer/tools/nixos-enter.sh [hunk 1]
```diff
@@ -60,6 +60,34 @@ chmod 0755 "$mountPoint/dev" "$mountPoint/sys"
 mount --rbind /dev "$mountPoint/dev"
 mount --rbind /sys "$mountPoint/sys"
 
+# modified from https://github.com/archlinux/arch-install-scripts/blob/master/arch-chroot.in#L26
```
Comment (roberth) [inline nixos/modules/installer/tools/nixos-enter.sh:63]:
```suggestion
# modified from https://github.com/archlinux/arch-install-scripts/blob/bb04ab435a5a89cd5e5ee821783477bc80db797f/arch-chroot.in#L26-L52
```

permalink

**Review by roberth — COMMENTED** [review]
Found some improvements, but would like someone more knowledgeable about the installer to make the decision.

File: nixos/modules/installer/tools/nixos-enter.sh (hunk 1 above)
Comment (Artturin) [reply nixos/modules/installer/tools/nixos-enter.sh]: Thanks

**Comment by Artturin** [comment]
rebased onto master and tested in virt-manager and ping works after in chroot by default

**Comment by SuperSandro2000** [comment]
I recently got hit by this. Thanks for changing this!

**Pull request merged**